packages/web/src/pages/App.js

```diff
@@ -466,52 +439,36 @@ class App extends Component {
         />
       )
 
-    const showBanner = !!(
-      showCTABanner ||
-      showWeb3ErrorBanner ||
-      showWebUpdateBanner
-    )
-
     const SwitchComponent = isMobile() ? AnimatedSwitch : Switch
     const noScroll = matchPath(this.state.currentRoute, CHAT_PAGE)
 
     return (
-      <DesktopHeaderContext.Provider value={{ offsetForBanner: showBanner }}>
-        {this.props.isChatEnabled ? <ChatListener /> : null}
-        <div className={cn(styles.app, { [styles.mobileApp]: isMobileClient })}>
-          {showCTABanner ? (
-            <MobileDesktopBanner
-              onClose={this.dismissCTABanner}
-              onAccept={this.showDownloadAppModal}
-            />
-          ) : null}
-          {showWeb3ErrorBanner ? (
-            <Web3ErrorBanner
-              alert
-              isElectron={client === Client.ELECTRON}
-              onClose={this.dismissWeb3ErrorBanner}
-            />
-          ) : null}
+      <div className={styles.root}>
+        <AppBannerWrapper>
+          <DownloadAppBanner />
+          <DirectMessagesBanner />
+          <Web3ErrorBanner />
+          {/* Other banners' logic is self-contained, but since this one uses the IPC 
```

Comment:
What happens if this banner is shown while the DMs banner is shown?

Reply:
They all stack in front of each other - whichever is the last one goes on top, and then they dismiss down to the lowest one.

We also have the option of making them stack vertically, though. I chose to keep the existing behavior

Reply:
Ooh, interesting. I imagine we probably only ever want to show one banner at a time, stacking them is weird - was the current behavior to stack?